Completely updated Swedish braille translation tables for grade 1, 2 …

InsynScandinavia opened the pull request:
…and 3.

These translation tables are created according to the standard published by Punktskriftsnämnden (the Swedish Braille Authority) in 2009.

bertfrees commented:
Thank you very much for your contribution! It's very exciting news that Swedish in Liblouis will finally get some love.

I have added it to milestone 3.23. That means we aim to include it in the September release. I would have liked to include it in the upcoming release of next Tuesday, but unfortunately I can see a few issues with the PR and I don't have any time to this weekend.

InsynScandinavia commented:
OK sounds great. Issues with PR? Can you elaborate?


Skickat från min iPhone

5 juni 2022 kl. 14:11 skrev Bert Frees ***@***.***>:

﻿

Thank you very much for your contribution! It's very exciting news that Swedish in Liblouis will finally get some love.

I have added it to milestone 3.23. That means we aim to include it in the September release. I would have liked to include it in the upcoming release of next Tuesday, but unfortunately I can see a few issues with the PR and I don't have any time to this weekend.

—
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#issuecomment-1146793404>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL6RZP3MDPW7SBA5F2TVNSKQLANCNFSM5X2RHICA>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

hammera commented:
Bert, I have possibility to help anybody this work related?
If I see right, all tests are passed right, this is a good starting point.
Easyer works I welcome offer help if a work related not need knoving the svedish language and svedish three grade braille standards, based with @InsynScandinavia coordination.

Unfortunatelly I not known the svedish language, But if your openion a good ydea, I welcome creating three testcase files, based with @InsynScandinavia defined rules (grade 1, grade 2 and grade 3 testcase files). So, I simple perhaps putting a text file the @InsynScandinavia added one contraction words and word rules, and I generate the three testcase corpus database. Of course this is a starting point testcase corpus database.

This is an easyer task, and possible doing easy automatic way, @InsynScandinavia only need review the three testcase files result.

With metadata informations related:
Now in liblouis/tables directory have an sv.tbl named file, with containing following informations:
```
#-index-name: Swedish
#-display-name: Swedish braille

#+language:sv
#+type:literary
#+grade:1

# TODO: Please correct the metadata above. It is not meant to be
# accurate nor complete. It hasn't been verified by the table
# author yet. It is merely an attempt by the liblouis maintainers
# to get some sensible initial values in place.

# TODO: Please add a reference to official documentation about
# the implemented braille code. Preferably submit the documents
# to https://github.com/liblouis/braille-specs.

include Se-Se-g1.utb
include braille-patterns.cti
```

Have an sv-1989.ctb named table, with based I think the BRLTTY svedish 8 dots computer braille table, and have an sv-1996 named table, with based the Swedish, computer, 1996 computer braille standard.

So, if I have any work related help possibility, I welcome offer help for you and @InsynScandinavia.

InsynScandinavia commented:
Hello Hammera,

I’m happy to help, obviously. 😊
Just let me know what I should look at.

I see that some of the languages seem to be missing a TBL file. I guess an updated one for Swedish grade 1 could look like something like this:

#-name: Swedish grade 1
#-index-name: Swedish, Sweden, uncontracted
#-display-name: Swedish uncontracted braille
#+language:sv
#+region:sv-SE
#+type:literary
#+contraction:no
#+grade:1
#+direction:both

Can someone please explain the differences between +name, +index-name and +display-name?

Regards,
David Renström
Insyn Scandinavia AB
Mossgatan 28
931 70 Skellefteå
010-455 04 20 / 070-588 39 20
E-post: ***@***.******@***.***>

[insyn-cmyk-logo NY]
Följ oss: Webb<http://www.insyn.se/> | Facebook<http://fb.com/insynab> | Twitter<http://twitter.com/insyn_se>
***@***.***

Från: hammera ***@***.***>
Skickat: den 7 juni 2022 09:54
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


Bert, I have possibility to help anybody this work related?
If I see right, all tests are passed right, this is a good starting point.
Easyer works I welcome offer help if a work related not need knoving the svedish language and svedish three grade braille standards, based with @InsynScandinavia<https://github.com/InsynScandinavia> coordination.

Unfortunatelly I not known the svedish language, But if your openion a good ydea, I welcome creating three testcase files, based with @InsynScandinavia<https://github.com/InsynScandinavia> defined rules (grade 1, grade 2 and grade 3 testcase files). So, I simple perhaps putting a text file the @InsynScandinavia<https://github.com/InsynScandinavia> added one contraction words and word rules, and I generate the three testcase corpus database. Of course this is a starting point testcase corpus database.

This is an easyer task, and possible doing easy automatic way, @InsynScandinavia<https://github.com/InsynScandinavia> only need review the three testcase files result.

With metadata informations related:
Now in liblouis/tables directory have an sv.tbl named file, with containing following informations:

#-index-name: Swedish

#-display-name: Swedish braille



#+language:sv

#+type:literary

#+grade:1



# TODO: Please correct the metadata above. It is not meant to be

# accurate nor complete. It hasn't been verified by the table

# author yet. It is merely an attempt by the liblouis maintainers

# to get some sensible initial values in place.



# TODO: Please add a reference to official documentation about

# the implemented braille code. Preferably submit the documents

# to https://github.com/liblouis/braille-specs.



include Se-Se-g1.utb

include braille-patterns.cti

Have an sv-1989.ctb named table, with based I think the BRLTTY svedish 8 dots computer braille table, and have an sv-1996 named table, with based the Swedish, computer, 1996 computer braille standard.

So, if I have any work related help possibility, I welcome offer help for you and @InsynScandinavia<https://github.com/InsynScandinavia>.

—
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#issuecomment-1148324160>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL5LK3SAERPRPMUUDUDVN35Z3ANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

hammera commented:
@InsynScandinavia, the table metatada informations not always put table maintainers a separate .tbl file, very often simple put the informations in top of the main language table.
For example because hungarian language have three table levels (eight dot computer Braille, grade 1 and grade 2 Braille), the hu-hu-comp8.ctb table putted we previous the metadata into the top of the table, the grade 1 table we doed a hu.tbl metadata file, and hu-hu-g2.ctb table we putted the hungarian grade2 level related informations metadata same in top of the table.

The metadata tags are describing into the [Table discovery based on table metadata Liblouis wiki page](https://github.com/liblouis/liblouis/wiki/Table-discovery-based-on-table-metadata#standard-metadata-tags), and found informations in Liblouis documentation  the [4. Table Metadata , 4.1 Syntax, 4.2 Query Syntax chapters](http://liblouis.org/documentation/liblouis.html#Table-Metadata).

Hopefully I linking good sections from the documentation.

hammera commented:
Following the diff output for tables/Makefile.am, I don't no possible attaching in Github a diff style patch or not, so copying into code blocks the patch:
```
diff --git a/tables/Makefile.am b/tables/Makefile.am
index 5fb7a050..3ac3389a 100644
--- a/tables/Makefile.am
+++ b/tables/Makefile.am
@@ -342,6 +342,9 @@ table_files = \
 	sv-1989.ctb \
 	sv-1996.ctb \
 	sv.tbl \
+	SV-SE-G1.utb \
+	SV-SE-G2.utb \
+	SV-SE-G3.utb \
 	ta.ctb \
 	tamil.cti \
 	ta-ta-g1.ctb \
```

@InsynScandinavia, I sent you the patch with attachment, feel free you commit it. I added your tables into tables/Makefile.am file.

bertfrees commented:
> I see that some of the languages seem to be missing a TBL file.

Like Attila said, just put the metadata in the SV-SE-*.utb tables. The .tbl files were a temporary solution for quickly adding (unverified) metadata to all existing tables.

Regarding file names, .utb is normally only used uncontracted braille tables, it's better to use .ctb for contracted braille tables.

> Can someone please explain the differences between +name, +index-name and +display-name?

It is explained in the Wiki page that Attila referred to. `#-name` is optional. It is the name of the table in the native language, Swedish. For `#-display-name` and `#-index-name`, check the file extra/generate-display-names/display-names which contains an overview of all the tables. `#-display-name` will be automatically checked so if it's not filled in correctly you will be notified :-).

Regarding the required changes in tables/Makefile.am, you may apply Attila's patch, but it's not mandatory. We'll fix it ourselves if needed.

What is missing from this PR and absolutely essential is some test data. Tests have two purposes:
- they serve as documentation of the implemented braille code
- they serve as a contract, they define how the table should behave: this helps with the table development and is a safety measure to avoid regressions in future version of Liblouis.

I will be putting some more comments within the code.

bertfrees commented on tables/SV-SE-G1.utb:
```diff
@@ -0,0 +1,506 @@
+# liblouis: Swedish Grade 1 Braille Table
+#
+#  Copyright (C) 2022 Insyn Scandinavia AB <david@insyn.se>
+#  Copyright (C) 2012 Mesar Hameed <mesar.hameed@gmail.com>
+#  Copyright (C) 2004-2008 ViewPlus Technologies, Inc. www.viewplus.com
+#  Copyright (C) 2004-2006 JJB Software, Inc. www.jjb-software.com
```
Could you explain in a few lines how this table came to be? It looks like you started with Se-Se-g1.utb (developed by the people in these copyright lines), and then updated it, to fix shortcomings or to match a new standard? You can maybe mention the project between Insyn Scandinavia and Punktskriftsnämnden that you mentioned in your email.

By the way you also mentioned in your email that you would not immediately replace Se-Se-g1.utb, but only remove it in a next release. I would replace it now, unless the difference between the two tables can be clearly explained. For example, maybe Se-Se-g1.utb implements an older version of the braille code that is still used. In that case we should keep Se-Se-g1.utb (and not remove it in the next Liblouis release).

bertfrees replied:
Thanks for the explanation and references.

I'll go ahead and remove Se-Se-g1.utb.

bertfrees commented on tables/SV-SE-G1.utb:
```diff
@@ -0,0 +1,506 @@
+lencapsphrase 3
+begcapsphrase 6-6-6
+endcapsphrase after 156
+sign \x00B7 3456			# so the line below compile
```
Is this line really needed?

bertfrees commented on tables/SV-SE-G1.utb:
```diff
@@ -0,0 +1,506 @@
+
+midendnumericmodechars .,\x00a0
+midnum \x00a0 3
+#numericnocontchars abcdefghij
```
I don't like commented out rules in tables so maybe you can add some explanation, e.g. something like: "the following lines could be replaced by a single numericnocontchars rule but it does not work because ...".

bertfrees commented on tables/SV-SE-G1.utb:
```diff
@@ -0,0 +1,506 @@
+include spaces.uti
+
+# Punctuation
+punctuation ! 235			exclamation mark		x0021
```
I know most tables include comments like this one ("exclamation mark") after character definitions, but actually such comments are mostly redundant and just clutter up the table. We're thinking of doing a big cleanup of unneeded comments in all existing tables. For new tables however maybe it is better to not include the comments in the first place.

bertfrees commented on tables/SV-SE-G1.utb:
```diff
@@ -0,0 +1,506 @@
+sign ² 4-6-126				superscript 2								x00B2
+sign ³ 4-6-146				superscript 3								x00B3
+sign ¹ 4-6-16					superscript 1								x00B9
+sign \x20AC 15-136-1235-135					Euro sign
```
For uniformity, I think it's better to replace escape sequences like \x20AC with the Unicode characters like €.

bertfrees commented on tables/SV-SE-G1.utb:
```diff
@@ -0,0 +1,506 @@
+midendnumericmodechars .,\x00a0
+midnum \x00a0 3
+#numericnocontchars abcdefghij
+nocontractsign 56
```
I release 3.22 we split up `nocontractsign` into [`nocontractsign`](http://liblouis.org/documentation/liblouis.html#nocontractsign-opcode) and [`nonumsign`](http://liblouis.org/documentation/liblouis.html#nonumsign-opcode). Please check if this rule still does what you want.

bertfrees commented:
@InsynScandinavia Does the Punktskriftsnämnden have documentation that we could link to from the tables? This creates clarity on what exactly the tables implement (which braille standard and which version of the standard).

bertfrees commented:
@InsynScandinavia Any chance we can include this in the September release? Have you looked at my comments?

InsynScandinavia commented:
Hi Bert,

I intend to work on the Swedish tables this weekend or next week. I absolutely would like to get them in for the September release, but I’ve been quite busy for a while.

Regards,
David Renström
Insyn Scandinavia AB
Mossgatan 28
931 70 Skellefteå
010-455 04 20 / 070-588 39 20
E-post: ***@***.******@***.***>

[insyn-cmyk-logo NY]
Följ oss: Webb<http://www.insyn.se/> | Facebook<http://fb.com/insynab> | Twitter<http://twitter.com/insyn_se>
***@***.***

Från: Bert Frees ***@***.***>
Skickat: den 11 augusti 2022 09:50
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


@InsynScandinavia<https://github.com/InsynScandinavia> Any chance we can include this in the September release? Have you looked at my comments?

—
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#issuecomment-1211658254>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL2ZK63AGK4JZHA7R6TVYSWELANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

InsynScandinavia commented:
Many of those commented lines were only kept for my own documentation what I had done. So I've now removed them.


Från: Bert Frees ***@***.***>
Skickat: den 17 juni 2022 10:28
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


@bertfrees commented on this pull request.

________________________________

In tables/SV-SE-G1.utb<https://github.com/liblouis/liblouis/pull/1226#discussion_r899892869>:

> +endcapsword 156

+lencapsphrase 3

+begcapsphrase 6-6-6

+endcapsphrase after 156

+sign \x00B7 3456                     # so the line below compile

+numsign 3456

+

+# Repetitions

+repeated ... 3-3-3            ellipsis

+repeated --- 36-36-36

+repeated ___ 45-36-45-36-45-36

+always \s--\s 36-36   tiret

+

+midendnumericmodechars .,\x00a0

+midnum \x00a0 3

+#numericnocontchars abcdefghij

I don't like commented out rules in tables so maybe you can add some explanation, e.g. something like: "the following lines could be replaced by a single numericnocontchars rule but it does not work because ...".

-
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#pullrequestreview-1010178559>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL2J2JWHZFK5RE5N2KLVPQZJVANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

InsynScandinavia commented:
No, it turns out this line is no longer needed and hence I've removed it.

Från: Bert Frees ***@***.***>
Skickat: den 17 juni 2022 10:26
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


@bertfrees commented on this pull request.

________________________________

In tables/SV-SE-G1.utb<https://github.com/liblouis/liblouis/pull/1226#discussion_r899890703>:

> +lowercase \x00E8 2346                      E with grave above     x00C8



+base uppercase É é



+lowercase é 123456                                                         E with acute above   x00C9



+base uppercase Ö ö



+lowercase ö 246                                                                                                                                                x00D6



+base uppercase Ü ü



+lowercase ü 1256                                            U with diaeresis               x00DC



+



+# Capital and number signs



+capsletter 6



+begcapsword 6-6                      # uncomment if you don't want capitalization



+endcapsword 156



+lencapsphrase 3



+begcapsphrase 6-6-6



+endcapsphrase after 156



+sign \x00B7 3456                     # so the line below compile



Is this line really needed?

-
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#pullrequestreview-1010175654>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL6JURGQXSWKL3NXKPLVPQZA3ANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

InsynScandinavia commented:
I haven't been able to test this until now, but it seems the new nonumsign opcode is working correctly for Swedish. So I have now changed the tables to use it instead.


Från: Bert Frees ***@***.***>
Skickat: den 17 juni 2022 10:46
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


@bertfrees commented on this pull request.

________________________________

In tables/SV-SE-G1.utb<https://github.com/liblouis/liblouis/pull/1226#discussion_r899908536>:

> +lencapsphrase 3

+begcapsphrase 6-6-6

+endcapsphrase after 156

+sign \x00B7 3456                     # so the line below compile

+numsign 3456

+

+# Repetitions

+repeated ... 3-3-3            ellipsis

+repeated --- 36-36-36

+repeated ___ 45-36-45-36-45-36

+always \s--\s 36-36   tiret

+

+midendnumericmodechars .,\x00a0

+midnum \x00a0 3

+#numericnocontchars abcdefghij

+nocontractsign 56

I release 3.22 we split up nocontractsign into nocontractsign<http://liblouis.org/documentation/liblouis.html#nocontractsign-opcode> and nonumsign<http://liblouis.org/documentation/liblouis.html#nonumsign-opcode>. Please check if this rule still does what you want.

-
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#pullrequestreview-1010201644>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOLZYKLEUSBP7YYSIAPDVPQ3NFANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

InsynScandinavia commented:
I have now included two links to the documentation in the headers.


Från: Bert Frees ***@***.***>
Skickat: den 17 juni 2022 10:51
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


@InsynScandinavia<https://github.com/InsynScandinavia> Does the Punktskriftsnämnden have documentation that we could link to from the tables? This creates clarity on what exactly the tables implement (which braille standard and which version of the standard).

-
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#issuecomment-1158650911>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL4AUHBBCSYBOOX5SO3VPQ4A5ANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

InsynScandinavia commented:
I have now uploaded YAML files for testing. They test for some issues that I have fixed but are not very extensive. I guess I'll expand the tests later on.


Från: Bert Frees ***@***.***>
Skickat: den 17 juni 2022 10:06
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


I see that some of the languages seem to be missing a TBL file.

Like Attila said, just put the metadata in the SV-SE-*.utb tables. The .tbl files were a temporary solution for quickly adding (unverified) metadata to all existing tables.

Regarding file names, .utb is normally only used uncontracted braille tables, it's better to use .ctb for contracted braille tables.

Can someone please explain the differences between +name, +index-name and +display-name?

It is explained in the Wiki page that Attila referred to. #-name is optional. It is the name of the table in the native language, Swedish. For #-display-name and #-index-name, check the file extra/generate-display-names/display-names which contains an overview of all the tables. #-display-name will be automatically checked so if it's not filled in correctly you will be notified :-).

Regarding the required changes in tables/Makefile.am, you may apply Attila's patch, but it's not mandatory. We'll fix it ourselves if needed.

What is missing from this PR and absolutely essential is some test data. Tests have two purposes:

  *   they serve as documentation of the implemented braille code
  *   they serve as a contract, they define how the table should behave: this helps with the table development and is a safety measure to avoid regressions in future version of Liblouis.

I will be putting some more comments within the code.

-
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#issuecomment-1158610110>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL5AUKJYGJ4C3EUPL6LVPQWWDANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

InsynScandinavia commented:
I have now added a paragraph about the braille table update and why it was needed in the headers.

I actually don't think that we need to keep any of the old Se-Se* files. I was a bit concerned that it would break things if we removed them, but I don't think that will be an issue.

Från: Bert Frees ***@***.***>
Skickat: den 17 juni 2022 10:25
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


@bertfrees commented on this pull request.

________________________________

In tables/SV-SE-G1.utb<https://github.com/liblouis/liblouis/pull/1226#discussion_r899890290>:

> +#  Copyright (C) 2012 Mesar Hameed ***@***.***>



+#  Copyright (C) 2004-2008 ViewPlus Technologies, Inc. www.viewplus.com



+#  Copyright (C) 2004-2006 JJB Software, Inc. www.jjb-software.com



Could you explain in a few lines how this table came to be? It looks like you started with Se-Se-g1.utb (developed by the people in these copyright lines), and then updated it, to fix shortcomings or to match a new standard? You can maybe mention the project between Insyn Scandinavia and Punktskriftsnämnden that you mentioned in your email.

By the way you also mentioned in your email that you would not immediately replace Se-Se-g1.utb, but only remove it in a next release. I would replace it now, unless the difference between the two tables can be clearly explained. For example, maybe Se-Se-g1.utb implements an older version of the braille code that is still used. In that case we should keep Se-Se-g1.utb (and not remove it in the next Liblouis release).

-
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#pullrequestreview-1010175110>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL4EOBYCRJ73XMJBEF3VPQY7RANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

InsynScandinavia commented:
I intend to do this later but at the moment I’m having troubles inserting many of the unicode characters in the editor, so I’ll leave them for now.


Från: Bert Frees ***@***.***>
Skickat: den 17 juni 2022 10:44
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


@bertfrees commented on this pull request.

________________________________

In tables/SV-SE-G1.utb<https://github.com/liblouis/liblouis/pull/1226#discussion_r899906062>:

> +sign | 456                                 vertical line                                 x007C



+sign ¢ 45-14                                 cent sign                                                    x00A2



+sign £ 45-123                        pound sign                                            x00A3



+sign ¤ 45-15                         currency sign                                 x00A4



+sign ¥ 45-13456                      yen                                                                          x00A5



+sign ○ 12456-1245     degrees



+sign ∙ 12456-3 multiplication



+sign § 346                                   paragraph                                                    x00A7



+sign © 236-14-356             copyright              x00A9



+sign ® 236-1235-356   registered trademark



+sign ™ 236-2345-134-356       trademark



+sign ° 4-356                                 degree sign                                                                 x00B0



+sign ² 4-6-126                       superscript 2                                                        x00B2



+sign ³ 4-6-146                       superscript 3                                                        x00B3



+sign ¹ 4-6-16                                superscript 1                                                        x00B9



+sign \x20AC 15-136-1235-135                                 Euro sign



For uniformity, I think it's better to replace escape sequences like \x20AC with the Unicode characters like €.

—
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#pullrequestreview-1010197275>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL2PRYQUYKBO6TRFF53VPQ3DVANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

bertfrees commented:
> I have now uploaded YAML files for testing. They test for some issues that I have fixed but are not very extensive. I guess I'll expand the tests later on.

Thanks very much. It's a start ;-)

I think I'm going to combine the three YAML files into a single sv.yaml file in tests/braille-specs/.

If it's OK I'll add the following at the top:

```yaml
# Swedish braille
#
# Based on the following documentation:
# - Svenska skrivregler för punktskrift <https://www.mtm.se/skrivregler/dtb-bm-l1.html>
# - Kortskrift nivå 1 och 2 <https://www.mtm.se/contentassets/08089b067f61495da1b035390d7a120e/kortskrift-niva-1-och-2---1997---nytt-manuskript-2016-09-26_v4.pdf>
#
# Copyright (C) 2022 by Insyn Scandinavia AB <david@insyn.se>
#
# Copying and distribution of this file, with or without modification,
# are permitted in any medium without royalty provided the copyright
# notice and this notice are preserved. This file is offered as-is,
# without any warranty.
```

If you agree I'll also add "Insyn Scandinavia AB <david@insyn.se>" to the AUTHORS file.

InsynScandinavia commented:
Sounds good.

Med vänliga hälsningar
David Renström
Insyn Scandinavia AB
Mossgatan 28
931 70 Skellefteå
010-455 04 20 / 070-588 39 20
E-post: ***@***.******@***.***>

[insyn-cmyk-logo NY]
Följ oss: Webb<http://www.insyn.se/> | Facebook<http://fb.com/insynab> | Twitter<http://twitter.com/insyn_se>
***@***.***

Från: Bert Frees ***@***.***>
Skickat: den 23 augusti 2022 22:03
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 0, 1 and 2 (PR #1226)


I have now uploaded YAML files for testing. They test for some issues that I have fixed but are not very extensive. I guess I'll expand the tests later on.

Thanks very much. It's a start ;-)

I think I'm going to combine the three YAML files into a single sv.yaml file in tests/braille-specs/.

If it's OK I'll add the following at the top:

# Swedish braille



#



# Based on the following documentation:



# - Svenska skrivregler för punktskrift <https://www.mtm.se/skrivregler/dtb-bm-l1.html>



# - Kortskrift nivå 1 och 2 <https://www.mtm.se/contentassets/08089b067f61495da1b035390d7a120e/kortskrift-niva-1-och-2---1997---nytt-manuskript-2016-09-26_v4.pdf>



#



# Copyright (C) 2022 by Insyn Scandinavia AB ***@***.***>



#



# Copying and distribution of this file, with or without modification,



# are permitted in any medium without royalty provided the copyright



# notice and this notice are preserved. This file is offered as-is,



# without any warranty.

If you agree I'll also add "Insyn Scandinavia AB ***@***.******@***.***>" to the AUTHORS file.

—
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#issuecomment-1224771428>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOLYVVU2PXABFCDXKOZTV2UU6FANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

bertfrees reviewed (changes requested):
OK I updated the branch and force-pushed it. I have a few comments left.

InsynScandinavia commented:
The additional meta tags look good. You’re right, the copyright notices were just copy-pasted into the tables for grade 1 and 2.
Can you please do the changes this time?

bertfrees commented:
OK I'll do it.

bertfrees commented on tables/sv-g0.utb:
```diff
@@ -0,0 +1,515 @@
+#+contraction: no
+#+grade: 0
+#+dots: 6
+#+direction: both
```
Oh, @InsynScandinavia one more thing: the table is marked as suited for both forward and backward translation, but there are only tests for forward translation.

bertfrees replied:
@InsynScandinavia Just making sure if you saw this question. If we could try wrapping up this PR before tomorrow that would be nice because I'm leaving on vacation. Christian will do the release on September 5 but we'd like to have everything ready well in time.

InsynScandinavia commented:
Thanks.

egli reviewed (approved):
Looks very good

InsynScandinavia commented:
Right, the tables support both forward and backward translation. However, since many characters use the same braille patterns, backtranslation isn’t successful in some instances.
What do you suggest? Should I create a specific YAML file for testing the backtranslation?

bertfrees commented:
> Should I create a specific YAML file for testing the backtranslation?

Yes, or better a new section for back-translation, in the same file. The `table` and `flags` keywords can appear more than once in a YAML file, so one file can combine tests for multiple tables and multiple directions.

Note that there is a `xfail` keyword that can be used to indicate that a test is currently expected to fail because Liblouis doesn't behave as the braille standard prescribes. (You can find lots of examples in other YAML files.)

egli commented:
Hm, something bad seems to have happened. The changes seem to be gone

bertfrees commented:
Indeed. @InsynScandinavia if you did it by accident, I can restore things for you.

InsynScandinavia commented:
Sorry, yes I did screw it up because I was too hasty. If you could restore, I would be grateful.

bertfrees commented:
Hmm I got a permission denied. Possible because the PR has been closed. So I think you'll have to do it yourself.

This should do the trick:

    git reset --hard  04f2f0f

Then you can push again.

InsynScandinavia commented:
I get the following:
fatal: ambiguous argument '04f2f0f': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

bertfrees commented:
OK I've pushed the commit to a temporary branch. Now you can do:

```
git fetch git@github.com:liblouis/liblouis.git pr-1226
git reset --hard  04f2f0f
```

InsynScandinavia commented:
I don’t seem to have permissions:

Warning: Permanently added 'github.com' (ED25519) to the list of known hosts.
***@***.******@***.***>: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

bertfrees commented:
Try using https://github.com/liblouis/liblouis.git.

InsynScandinavia commented:
Since I don’t wanna mess up again, I get a failed merge with NEWS, what should I do with it? rm or add? I’m sorry, I’m not used to git but use Perforce almost every day.

bertfrees commented:
Hum. What do you want to merge? You can just add more commits onto 04f2f0f. If any merging is required we take care of that.

bertfrees commented:
Resolving merge conflicts in NEWS usually means keeping the two entries that were added (one in each branch).

InsynScandinavia commented:
Ok, I’ll remove NEWS here.
Another issue: I cannot run lou_checkyaml since the sv tables are not known to it. What to do now?

bertfrees commented:
The test file needs to be added to tests/Makefile.am and tests/braille-specs/Makefile.am. If you push the branch again and reopen the PR, I can do it.

InsynScandinavia commented:
You had already done this, it seems. Thanks. At least sv.yaml is listed in both. I see that you have updated and improved sv.yaml as well. Could we just accept it as it is for now? I can improve it later. I’ve tried to build the Windows version for a couple of hours now, but I keep getting linker errors, i.e. 12 unresolved symbols which seem to be related to msvcrt.lib. Have tried different versions of this library without any luck. I don’t know if this is a specific issue for Visual Studio Community 2022 or if something else is wrong on my system.

InsynScandinavia commented:
I’ve now tried compiling on Linux (Debian) as well. When running autoconf, I get many errors about possibly undefined symbols. The output ./configure also generates an error.

bertfrees commented:
> When running autoconf, I get many errors about possibly undefined symbols. The output ./configure also generates an error.

Do you run `./autogen.sh` first?

InsynScandinavia commented:
I have now removed all those character comments.

Från: Bert Frees ***@***.***>
Skickat: den 17 juni 2022 10:42
Till: liblouis/liblouis ***@***.***>
Kopia: David Renstrom ***@***.***>; Mention ***@***.***>
Ämne: Re: [liblouis/liblouis] Completely updated Swedish braille translation tables for grade 1, 2 and 3 (PR #1226)


@bertfrees commented on this pull request.

________________________________

In tables/SV-SE-G1.utb<https://github.com/liblouis/liblouis/pull/1226#discussion_r899904364>:

> +#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU

+#  Lesser General Public License for more details.

+#

+#  You should have received a copy of the GNU Lesser General Public

+#  License along with liblouis. If not, see

+#  <http://www.gnu.org/licenses/>.

+

+# Characters not to backtranslate (must be included for braille input to work)

+include braille-patterns.cti

+

+# Spaces

+space \x00a0 0

+include spaces.uti

+

+# Punctuation

+punctuation ! 235                    exclamation mark               x0021

I know most tables include comments like this one ("exclamation mark") after character definitions, but actually such comments are mostly redundant and just clutter up the table. We're thinking of doing a big cleanup of unneeded comments in all existing tables. For new tables however maybe it is better to not include the comments in the first place.

-
Reply to this email directly, view it on GitHub<https://github.com/liblouis/liblouis/pull/1226#pullrequestreview-1010194877>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZPGOL2MO3ISC36CDEJSW6TVPQ24BANCNFSM5X2RHICA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

InsynScandinavia commented:
Thanks, it seems to have worked. I’ll get back to you ASAP.

InsynScandinavia commented:
OK, I’ll take a look tonight. I guess I need to compile with MinGW on Windows? I tried it yesterday but I couldn’t install it on my system because of Cylance which blocks just about everything. But I’ll use another system for it.

bertfrees commented:
@InsynScandinavia Not sure which comments you are replying to.

I have no experience with building on Windows myself, but yes MinGW is an option I think. If you also have a Linux machine I would recommend that though.
